core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java

```diff
@@ -91,27 +68,44 @@ public void close() {
         try {
             ((CloseableHttpClient)getHttpClient()).close();
         } catch (IOException e) {
-            InternalLogger.INSTANCE.error("Failed to close http client, exception: %s", e.toString());
+            InternalLogger.INSTANCE.error("Failed to close http httpClient, exception: %s", e.toString());
+        } finally {
```

Comment:
It seems this is a redundant locking. The methods that call this are synchronized already. My recommendation would be to prevent locks wherever possible.

Reply:
It's good practice to keep a class thread-safe within itself. I don't want to rely on the current call stack to ensure thread safety. Since there won't be any other threads calling this, there won't be any contention. It's simply a precautionary statement. It won't affect performance.

Reply:
Sounds good.